debian/opensesame.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.40.4.
.TH 0.27 "1" "December 2012" "0.27 'Frisky Freud'" "User Commands"
.SH NAME
0.27 \- Graphical experiment builder for the social sciences
.SH SYNOPSIS
.B opensesame
[\fIexperiment\fR] [\fIoptions\fR]
.SH OPTIONS
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.IP
Immediately run an experiment:
.TP
\fB\-r\fR, \fB\-\-run\fR
Run fullscreen
.TP
\fB\-w\fR, \fB\-\-run\-in\-window\fR
Run in window
.IP
Miscellaneous options:
.TP
\fB\-c\fR _CONFIG, \fB\-\-config\fR=\fI_CONFIG\fR
Set a configuration option, e.g, '\-\-config
auto_update_check=False;scintilla_font_size=10'. For a
complete list of configuration options, please refer
to the source of config.py.
.TP
\fB\-t\fR _THEME, \fB\-\-theme\fR=\fI_THEME\fR
Specify a GUI theme
.TP
\fB\-d\fR, \fB\-\-debug\fR
Print lots of debugging messages to the standard
output
.TP
\fB\-s\fR, \fB\-\-stack\fR
Print stack trace (only in debug mode)
.TP
\fB\-p\fR, \fB\-\-preload\fR
Preload Python modules
.TP
\fB\-\-pylink\fR
Load PyLink before PyGame (necessary for using the
Eyelink plug\-ins in non\-dummy mode)
.TP
\fB\-\-ipython\fR
Enable the IPython interpreter
.TP
\fB\-\-locale\fR
Specify localization
.TP
\fB\-\-catch\-translatables\fR
Log all translatable text
.TP
\fB\-\-no\-global\-resources\fR
Do not use global resources on *nix


CHANGELOG
For changes after 0.25, please refer to GitHub

0.24 "Cody Crick"
- Added back-end independence
- Added experimental PsychoPy backend (psycho)
- Added experimental OpenGL backend (psycho)
- Added Python functions to the loop item
- Improved interface for the loop item
- Improved startup time under windows

0.23 "Blue Beck"
- Fixed a bug where no items could be added if the main sequence was replaced
- Fixed a bug where the autosave timer continued while the experiment was running
- Replace pygame icon by custom window icon for the experiment window
- The keyboard is now flushed before every sequence and with every loop cycle, to catch escape presses
- Added a replace dialog to the inline_editor
- The keyboard is now flushed during synth and sampler playback, to catch escape presses
- Tabs are now draggable
- Script changes are now automatically applied when running/ saving the experiment
- Renamed "Tools" to "Options" in menu
- Added immediate rename option, to rename items as soon as they are created (default = off)
- Fixed openexp.experiment.sound_init() to be compatible with older PyGame versions (tested on 1.8.1)
- Added sound_sample_size and sound_channels to sound settings in openexp.experiment
- Switched to optparse module for parsing command line arguments
- Added opensesamerun to run experiments standalone
- Added unicode support to sketchpads to allow for special characters
- Fixed a bug where applying a script with special characters caused an error
- Added the run as separate process option to the opensesame gui
- Added slider control to qtplugin
- Made graphics mode configurable
- Added preferences tab and removed some of the options from the menu
- Fixed a bug where changes to the "Run if" statement weren't automatically applied
- Added splash screen
- Added video files as known file types to the file pool
- Added recent files dialog
- Added editor to the qtplugins automatic controls
- Byte-compile all inline_scripts, conditional statements, etc.
- Added indent/ unindent buttons to the inline editor

0.22 "Axiomatic Axelrod"
- Added search capabilities to inline_editor and a notification of unapplied changes
- Added noise patch as a drawing primitive to the sketchpad
- Fixed a bug where the sketchpad gave errors if it failed to parse a draw command
- Fixed a bug where pre-release versions broke update checking
- Fixed a bug where changing the working directory only worked if the path didn't contain spaces
- The default location for the logfile is now the location of the experiment
- Print stacktrace on unexpected exceptions
- Doesn't ask for subject nr, logfile etc. in debug mode
- Added an optional hack (--pylink) to make sure that pylink is load before pygame (if available)
- Added qtplugin.add_text() function
- Fixed a bug in var_info() which caused the logger to crash on advanced_delay and text_input plugins 
- Added cleanup functions to the experiment, to cleanup afterwards
- Added restart option, to circumvent pylinks refusal to deinitialize (disabled by default, since the problems seems to be fixed)
- Fixed a papercut in the logger which caused unpredictable ordering of columns
- Fixed a bug which could cause the sketchpad to crash on incorrectly formatted draw items, causing a "variable g not defined error".
- Fixed a bug where exceptions on apply script weren't caught and caused a crash
- Fixed a bug where changing the "Show if" option in sketchpads wasn't automatically applied
- Added a delete option to the context menu in the overview tree of items in a sequence
- Added comments to OpenSesame script, indicated by # or //
- Sketchpads now throw an exception if they can't parse a command instead of simply ignoring it
- Added Alt+A apply shortcut to script editor
- Fixed a papercut where repeat couldn't be higher than 100 in a loop item
- Added external_script plug-in
- Started using codenames for releases
- Massively improved tables in loop items
- Created a specific help_browser, which allows clicking on hyperlinks and shows a footer message
- Changes about from a dialog to a help tab
- Added contribute and bug submission help tabs

0.21
- Fixed a bug where text consisting of only digits caused the sketchpad to crash
- Fixed a papercut where using scaled images in a sketchpad caused a deprecation warning
- Changed logfile dialog to indicate escape key
- Added open on double-click to file pool
- Changed the file pool automatic renaming notification
- Added different error messages for "image not found" and "unsupported image format" to the sketchpad
- Made the error message for non-existent variabled in a logger more transparent
- Made .opensesame.tar.gz the default format
- Fixed a papercut where the logger jumped to the top of the tab after a variable was (de)selected
- Fixed a papercut where the logger displayed the same variable twice if it was offered by two items (or more)
- Fixed a bug where the loop order was reset to random on opening a file
- Loop items now store the order of the columns in the table
- Added copy/paste keyboard shortcuts to loop item table
- Fixed a bug where some changes to the loop tab weren't applied to the script immediately
- Keep track of unsaved changes and ask if they should be saved on exit
- Make use of the statusbar
- Added auto-save function for automatic creation of backups
- Fixed a bug which crashed the sampler on changing the duration to a numeric value
- Added multiple select and remove to file pool
- The old name is now suggested on renaming files from the file pool
- Ask before overwriting files in the file pool
- Added inline_script.copy_sketchpad() convenience function
- Sort items by type and name in pulldown menus
- Close tabs on deleting unused items to prevent non-existent items having a tab
- Added possibility to rename from the sequence tab
- Removed the inconsistent "Draw a line" tooltip from the sketchpad
- Added thick gridlines to indicate the display center in the sketchpad

opensesame.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-

"""
This file is part of OpenSesame.

OpenSesame is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

OpenSesame is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with OpenSesame.  If not, see <http://www.gnu.org/licenses/>.
"""

if __name__ == "__main__":

	# First, load a minimum number of modules and show an empty app window. This
	# gives the user the feeling of a snappy response.
	import os, sys, platform

	# Change the working directory on Windows. Depending on whether the
	# application has been frozen by py2exe or not we need to use a different
	# method of deducing the name to the main script.
	# See also http://www.py2exe.org/index.cgi/HowToDetermineIfRunningFromExe
	if os.name == "nt":
		import imp
		if (hasattr(sys, "frozen") or hasattr(sys, "importers") or \
			imp.is_frozen("__main__")):
			path = os.path.dirname(sys.executable)
		else:
			path = os.path.dirname(__file__)
		if path != '':
			os.chdir(path)
			if path not in sys.path:
				sys.path.append(path)

	# horrendous OS X invalid locale hack, which otherwise prevents psychopy to start
	if platform.system() == 'Darwin':
		import locale
		if locale.getlocale()[0] is None:
			locale.setlocale(locale.LC_ALL, 'en_US.UTF-8')
			
		# On 32-bit OSX: Shortly init and destroy pygame window before all else
		# Strangly, the experiment otherwise hangs at init_dispay() with pygame based backends...
		if not sys.maxsize > 2**32:
			import pygame
			pygame.init()
			pygame.display.init()
			pygame.display.set_mode((10,10))
			pygame.quit()	
				
	# Load debug package (this must be after the working directory change)
	from libopensesame import debug

	# The IPython modules must be loaded before PyQt4
	if "--ipython" in sys.argv:
		try:
			from IPython.frontend.qt.console.ipython_widget import IPythonWidget
			debug.msg('IPython loaded')
		except:
			sys.argv.remove('--ipython')
			debug.msg('Failed to load IPython. Is IPython QtConsole installed?', \
				reason='warning')

	# Do the basic window initialization
	from PyQt4.QtGui import QApplication
	app = QApplication(sys.argv)
	from libqtopensesame.qtopensesame import qtopensesame
	opensesame = qtopensesame(app)		
	app.processEvents()
	
	# Import the remaining modules
	from PyQt4.QtCore import SIGNAL, QObject, QLocale, QTranslator	
	from libopensesame.misc import resource
	import os.path
	
	# Load the locale for UI translation. The locale can be specified on the
	# command line using the --locale parameter
	locale = unicode(QLocale().system().name())
	for i in range(len(sys.argv)-1):
		if sys.argv[i] == '--locale':
			locale = sys.argv[i+1]	
	qm = resource(os.path.join('locale', locale) +'.qm')
	if qm != None:
		debug.msg('installing %s translator' % qm)
		translator = QTranslator()
		translator.load(qm)
		app.installTranslator(translator)
	else:
		debug.msg('no translator found for %s' % locale)	

	# Now that the window is shown, load the remaining modules and resume the
	# GUI initialization.
	opensesame.resume_init()
	QObject.connect(app, SIGNAL("quit()"), opensesame.close)
	
	# Open an experiment if it has been specified as a command line argument and
	# suppress the new wizard in that case.
	import os.path
	if len(sys.argv) >= 2 and os.path.isfile(sys.argv[1]):
		start_new_tab = False
		opensesame.open_file(path=sys.argv[1])
	else:
		start_new_tab = True

	# Run the app (the startup method depends on command line parameters)
	if opensesame.options.run:
		opensesame.run_experiment()
	elif opensesame.options.run_in_window:
		opensesame.run_experiment(False)
	else:
		opensesame.check_update(always=False)
		if start_new_tab:
			opensesame.ui.tabwidget.open_start_new(start=True)

	opensesame.restore_window_state()				
	opensesame.refresh()			
	opensesame.show()
	opensesame.raise_()  #Added for OS X, otherwise Window will not appear
	
	# Exit using the application exit status
	sys.exit(app.exec_())

